tests/phpunit/includes/build-equivalent-html-semantic-tree.php

```diff
@@ -0,0 +1,299 @@
+<?php
+
+/* phpcs:disable WordPress.Security.EscapeOutput.ExceptionNotEscaped */
+
+/**
+ * Generates the tree-like structure represented in the Html5lib tests.
+ *
+ * That format is extended with a special representation of block delimiters and their attributes.
+ * Furthermore, it the order of attributes and class names is normalized both for HTML tags and block delimiters,
+ * as is the whitespace in HTML tags' style attribute.
```

Comment:
Some suggestions to:

- Make the function short description more clear and to the point. html5lib isn't relevant to the essential behavior of this function.
- Improve some language in the description around blocks and talk about block semantics instead of delimiters.

These lines may need to be properly formatted.

```suggestion
 * Generates representation the semantic HTML tree structure. 
 *
 * This is inspired by the representation used by the HTML5lib tests. It's been extended here for 
 * blocks to render the semantic structure of blocks and their attributes.
 * The order of attributes and class names is normalized both for HTML tags and blocks,
 * as is the whitespace in HTML tags' style attribute.
```